packages/cozy-doctypes/src/Folder.js

```diff
@@ -0,0 +1,54 @@
+const Application = require('./Application')
+const CozyFile = require('./File')
+
+/**
+ * Class representing the folder model.
+ * @extends CozyFile
+ */
+class CozyFolder extends CozyFile {
+  static async getReferencedFolders(client, reference) {
+    const { included } = await client
+      .collection(CozyFile.doctype)
+      .findReferencedBy({
+        _type: Application.doctype,
+        _id: reference
+      })
+    return included.filter(
+      folder => !/^\/\.cozy_trash/.test(folder.attributes.path)
+    )
+  }
+
+  static async getOrCreateFolderWithReference(client, path, reference) {
+    const existingFolders = await CozyFolder.getReferencedFolders(
+      client,
+      reference
+    )
+
+    if (existingFolders.length) return existingFolders[0]
+
+    const collection = client.collection(CozyFile.doctype)
+    const dirId = await collection.ensureDirectoryExists(path)
+    await collection.addReferencesTo(
+      {
+        _id: reference,
+        _type: Application.doctype
+      },
+      [
+        {
+          _id: dirId
+        }
+      ]
+    )
+
+    const { data: dirInfos } = await collection.get(dirId)
+
+    return dirInfos
+  }
+}
+
+CozyFolder.REF_ADMINISTRATIVE = `${Application.doctype}/administrative`
+CozyFolder.REF_PHOTOS = `${Application.doctype}/photos`
+CozyFolder.REF_UPLOAD = `${Application.doctype}/photos/upload`
+CozyFolder.REF_BACKUP = `${Application.doctype}/photos/mobile`
```

Comment:
I personally do not understand why REF_BACKUP is not `${Application.doctype}/backup` ? What is this folder supposed to be ?

Reply:
This is a question for @Crash-- or @y-lohse I guess. This code is directly extracted from Drive.

Reply:
because it has to be specific. We only backup the photos on mobile, so photos/mobile is more pertinent & relevant than /backup. `backup` is not precise enough, it can be the folder backuped by drive desktop or else. 

REF_BACKUP should / can be renamed to REF_PHOTO_BACKUP since we're not in a Drive / Photos context anymore.